docs/contracts/NFT Contract/NFT Actions/purchase.md

```diff
@@ -0,0 +1,91 @@
+---
+title: 'purchase'
+order: 31
+deploy: ['experimental']
+---
+
+# Purchase
+
+This action is used to purchase uniqs directly from a token factory.
+
+## Technical Behavior
+
+TBA
+
+## Action Parameters
+
+**Action Interface**
+
+| Property Name       | C++ Type                       | JavaScript Type       |
+| ------------------- | ------------------------------ | --------------------- |
+| token_factory_id    | uint64_t                       | number                |
+| index               | uint64_t                       | number                |
+| max_price           | asset                          | string                |
+| buyer               | optional name                  | string / null         |
+| receiver            | optional asset                 | string / null         |
+| promoter_id         | optional name                  | string / null         |
+| user_uniqs          | optional provided_user_uniqs   | Array / null          |
```

Comment:
For optional types you could still put angled braces if you put the \ in front of them like optional\<name>